ambry-store/src/main/java/com/github/ambry/store/BlobStore.java

```diff
@@ -512,9 +514,17 @@ public void delete(List<MessageInfo> infosToDelete) throws StoreException {
       List<IndexValue> indexValuesToDelete = new ArrayList<>();
       List<Short> lifeVersions = new ArrayList<>();
       Offset indexEndOffsetBeforeCheck = index.getCurrentEndOffset();
+      maybeCallBetweenGetEndOffsetAndFindKeyFunc();
```

Comment:
Since we have this callable here, can we remove `maybeCallBeforeSynchronizationFunc` in this delete method. I think we can reuse `maybeCallBetweenGetEndOffsetAndFindKeyFunc` to complete other concurrency tests.

Reply:
I think they are both necessary. They are testing different corner cases. And I certainly want to remove the getEndOffsetAndFindKey callback after in next PR.